src/test/java/com/jcabi/github/mock/MkHookTest.java

```diff
@@ -107,26 +139,28 @@ public void createWithCorrectRepo() throws Exception {
      * Test if {@link MkHook} is being created with the correct id.
      * @throws Exception If something goes wrong
      */
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     public void createWithCorrectId() throws Exception {
         MatcherAssert.assertThat(
             "Hook json returned wrong id",
             this.hook.json().getInt("id"),
-            new IsEqual<Integer>(Tv.FIVE)
+            new IsEqual<Integer>(
+                Integer.parseInt(this.xml.xpath("id/text()").get(0))
+            )
         );
     }
 
     /**
      * Test if {@link MkHook} is being created with the correct url.
      * @throws Exception If something goes wrong
      */
-    @Test(expected = UnsupportedOperationException.class)
+    @Test
     public void createWithCorrectUrl() throws Exception {
         MatcherAssert.assertThat(
             "Hook json returned wrong url",
             this.hook.json().getString("url"),
             new IsEqual<String>(
-                "https://api.github.com/repos/jcabi/jcabi-github/hooks/1"
+                this.xml.xpath("url/text()").get(0)
```

Comment:
@llorllale Again, we should test if it's returning the string we set before, not a value searched from the xml.

Reply:
@paulodamaso what we "set" is an XML with a structure that `MkHook` must know and understand. This XML is given to `MkHook` by `MkHooks`. So, we must test that `MkHook` understands this XML. It doesn't really matter what values we use; what matters is that `MkHook#json()` agrees with what's in the XML.

As an aside: I think this relationship between *Mk<iterable>* (eg `MkHooks`) and *Mk<singular>* (eg `MkHook`) should be inverted. The XML is an implementation detail that only `MkHook` should know about. If that were the case, then I think structuring the tests in the way you suggest would be more applicable.

Reply:
@llorllale I still think that it's not a good idea to use our tests to "double test" our classes (testing the same thing twice, like `this.hook.json().getString("url")` and `this.xml.xpath("url/text()").get(0)`); actually, we must test if our objects are being created with the correct values, and not if the both method are returning the same value (if they both perform wrongly the test can still pass), so that's why I consider importing that we use known and defined values for testing in this case.

About the relationship between our classes, I think that in this case it's ok. This `iterable-singular` relationship is indeed high-coupled because `MkHook` must know from where to read the data for its operations and `MkHooks` must know from where to fetch and create new `MkHook` objects, so I think there's no big deal in using them that way.

The problem that I see is that we shouldn't be operating over our objects internal XML neither retrieving it as json: both of them kills encapsulation. After all, we have a lot of naked data around  the project because we set values in xml and in json everywhere.

Reply:
@paulodamaso I don't agree at all. This is not "double-testing".

I was following the direction you set for these tests: couple them all to a single `MkHook` instance. If I do what you want now, then I'll also have "magic values" inside test methods.

I just browsed through some unit test files and I don't see tests coupled to a single test object, so I think that's the first mistake in these tests for `MkHook`.

I will refactor a bit later on. Once I do, I will have the reference values explicit within each test method. Would that satisfy you?

Reply:
@llorllale Yes, it will, thanks. About the "magic numbers", I'm not against using them in tests; actually, we must use them for equivalence partitioning testing and boundary-value analysis (in this case, our "partition" and "boundaries" are composed of exactly one value, that is the value we expect from the test).